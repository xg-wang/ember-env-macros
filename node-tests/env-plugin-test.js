/* eslint-env jest */
const { transformSync } = require('@babel/core');
const EnvPlugin = require('../lib/env-plugin');

test('works', () => {
  const { code } = transformSync(
    `import { configEnv, buildEnv } from 'ember-env-macros';
configEnv('string')
configEnv('number')
configEnv('boolean')
buildEnv('string')
buildEnv('number')
buildEnv('boolean')`,
    {
      plugins: [
        [
          EnvPlugin,
          {
            configEnv: {
              string: 'config',
              number: 1,
              boolean: true,
            },
            buildEnv: {
              string: 'build',
              number: 2,
              boolean: false,
            },
          },
        ],
      ],
    }
  );
  expect(code).toMatchSnapshot();
});

test('dot separeted key', () => {
  const { code } = transformSync(
    `import { configEnv, buildEnv } from 'ember-env-macros';
configEnv('some.key')
buildEnv('another.key')`,
    {
      plugins: [
        [
          EnvPlugin,
          {
            configEnv: {
              some: {
                key: 'value',
              },
            },
            buildEnv: {
              another: {
                key: 'eulav',
              },
            },
          },
        ],
      ],
    }
  );
  expect(code).toMatchSnapshot();
});

test('default value', () => {
  const { code } = transformSync(
    `import { configEnv, buildEnv } from 'ember-env-macros';
configEnv('some.missingKey', 'defaultConfig')
buildEnv('anotherMissing', 'defaultBuild')`,
    {
      plugins: [
        [
          EnvPlugin,
          {
            configEnv: {
              some: {
                key: 'value',
              },
            },
            buildEnv: {
              another: {
                key: 'eulav',
              },
            },
          },
        ],
      ],
    }
  );
  expect(code).toMatchSnapshot();
});
