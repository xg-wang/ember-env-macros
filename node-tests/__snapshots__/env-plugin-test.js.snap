// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default value 1`] = `
"\\"defaultConfig\\";
\\"defaultBuild\\";"
`;

exports[`dot separeted key 1`] = `
"\\"value\\";
\\"eulav\\";"
`;

exports[`works 1`] = `
"\\"config\\";
1;
true;
\\"build\\";
2;
false;"
`;
